src/java.desktop/share/classes/java/beans/DefaultPersistenceDelegate.java

```diff
@@ -197,8 +197,7 @@ private void doProperty(Class<?> type, PropertyDescriptor pd, Object oldInstance
             if (!Objects.equals(newValue, out.get(oldValue))) {
                 // Search for a static constant with this value;
                 Object e = (Object[])pd.getValue("enumerationValues");
-                if (e instanceof Object[] && Array.getLength(e) % 3 == 0) {
-                    Object[] a = (Object[])e;
+                if (e instanceof Object[] a && Array.getLength(e) % 3 == 0) {
```

Comment:
Something strange is going on here. In previous line we casted expression to `Object[]`, but assigned to `Object` type